agent_id/src/AgentID.cpp

```diff
@@ -71,16 +73,28 @@ void AgentID::operator=(const std::vector<uint8_t>& other_id)
 
 AgentID::operator uint64_t() const
```

Comment:
Would be useful to mention in the in the header that this function throws an error, so that IDEs can show what to expect when we use this function